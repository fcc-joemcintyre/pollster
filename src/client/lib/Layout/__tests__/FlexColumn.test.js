/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { FlexColumn } from '..';

test ('FlexColumn renders default gap', () => {
  const component = renderer.create (
    <FlexColumn>
      <button type='submit'>SUBMIT</button>
      <button type='button'>CANCEL</button>
      <button type='button'>CLEAR</button>
    </FlexColumn>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('FlexGroup renders with custom gap', () => {
  const component = renderer.create (
    <FlexColumn gap='10px'>
      <button type='submit'>SUBMIT</button>
      <button type='button'>CANCEL</button>
      <button type='button'>CLEAR</button>
    </FlexColumn>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
