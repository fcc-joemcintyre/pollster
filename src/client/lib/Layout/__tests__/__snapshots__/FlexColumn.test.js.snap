// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexColumn renders default gap 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * {
  margin-top: 20px;
}

.c0 > *:first-child {
  margin-top: 0;
}

<div
  className="c0"
>
  <button
    type="submit"
  >
    SUBMIT
  </button>
  <button
    type="button"
  >
    CANCEL
  </button>
  <button
    type="button"
  >
    CLEAR
  </button>
</div>
`;

exports[`FlexGroup renders with custom gap 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * {
  margin-top: 10px;
}

.c0 > *:first-child {
  margin-top: 0;
}

<div
  className="c0"
>
  <button
    type="submit"
  >
    SUBMIT
  </button>
  <button
    type="button"
  >
    CANCEL
  </button>
  <button
    type="button"
  >
    CLEAR
  </button>
</div>
`;
